### DISCUSSION TITLE
For evaluation, how could I create a private chat channel?

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#921] [Q&A] [answered]
I'm trying to figure out how the built-in API Rules could be used to make a real time chat server.  Is it possible to put the rules such that they require the requesting user to be a linked to in another collection, or something else?  Is there a way to get ahead of the route being executed to add custom Golang logic?  I see `OnRecordsListRequest`, but that seems to happen after the query.  I'm using the "authorized private chat room" example because it seems to be the most simple example that I can come up with.

Or perhaps I'm thinking about the issue in the wrong paradigm.  What is the idiomatic way for Pocketbase to segment a collection by security?

**Comment by ganigeorgiev** [comment, marked as answer]
> Is it possible to put the rules such that they require the requesting user to be a linked to in another collection, or something else?

Yes. In PocketBase v0.8.x, your app users are just regular collection records and you can create a `relation` field pointing to it. For example:
```
@request.auth.id = someRelationField.anotherRelationField.user
```

(for PocketBase v0.7.x is similar but there you have 1 additional options because `users` and `profiles` are separate entities and in v0.8 they are now merged by default per #376)

The documentation is still not updated for v0.8.0, but if you haven't already you can find more info about the API rules in https://pocketbase.io/docs/manage-collections/#collection-api-rules

---

>  Is there a way to get ahead of the route being executed to add custom Golang logic?

The list and view action currently doesn't have a dedicated `before` and `after` hook since I'm not sure how useful that would be, but we can add it if there is users demand for that. Alternatively, you can also create a global `echo` middleware(s) by registering it inside the `OnBeforeServe` hook and intercept the request based on its path before reaching the route handler.
